lib/api/apiUtils/object/expirationHeaders.js

```diff
@@ -37,8 +37,10 @@ const AMZ_ABORT_ID_HEADER = 'x-amz-abort-rule-id';
 
 function _generateExpHeadersObjects(rules, params, datetime) {
     const tags = {
-        TagSet: Object.keys(params.tags)
-            .map(key => ({ Key: key, Value: params.tags[key] })),
+        TagSet: params.tags
+            ? Object.keys(params.tags)
+                .map(key => ({ Key: key, Value: params.tags[key] }))
```

Comment:
Small enhancement suggestion, not related to this PR's change (feel free to ignore):

```suggestion
            ? Object.entries(params.tags)
                .map(([key, value]) => ({ Key: key, Value: value }))
```